tests/sanity/sanity_test.go

```diff
@@ -55,16 +54,53 @@ var _ = AfterSuite(func() {
 })
 
 var _ = Describe("AWS FSx for OpenZFS CSI Driver", func() {
-	_ = os.MkdirAll("/tmp/csi", os.ModePerm)
-	config := &sanity.Config{
-		Address:        endpoint,
-		TargetPath:     mountPath,
-		StagingPath:    stagePath,
-		TestVolumeSize: 2000 * util.GiB,
-		TestVolumeParameters: map[string]string{
+	Context("Filesystem", Ordered, func() {
+		config := sanity.NewTestConfig()
+		config.Address = endpoint
+		config.TestVolumeParameters = map[string]string{
 			"ResourceType":              "filesystem",
-			"SkipFinalBackupOnDeletion": `true`,
-		},
-	}
-	sanity.GinkgoTest(config)
+			"SkipFinalBackupOnDeletion": "true",
+		}
+
+		BeforeAll(func() {
+			fsxOpenZfsDriver.ResetCloud()
+		})
+
+		Describe("CSI sanity", func() {
+			BeforeEach(func() {
+				switch {
+				case strings.Contains(CurrentSpecReport().FullText(), "should create volume from an existing source snapshot"),
```

Comment:
Discussed offline - the tests that we're excluding here are some of the tests that were newly added as part of `github.com/kubernetes-csi/csi-test/v5/pkg/sanity`. Some are not applicable to our implementation (e.g. snapshots of file systems), while others should be caught at the Kubernetes/sidecar layer. Adding another layer to protect against these situations here would require a large effort and involve making additional API calls. I think these are fine to exclude.